### PR TITLE
CNOT implementation as edge property

### PR DESCRIPTION
**Pull request by dieris**
Something simple for https://github.com/BBN-Q/QGL/issues/136? With bonus camel_case crusade part 2 https://github.com/BBN-Q/QGL/commit/7ad56f90e0094138cd5e2fe2014cd650f3f25a27

### CODE REVIEW

File: QGL/Channels.py [hunk 1]
```diff
@@ -195,10 +195,11 @@ class Edge(LogicalChannel):
                                 'cutoff': 2,
                                 'drag_scaling': 0,
                                 'sigma': 5e-9,
-                                'riseFall': 20e-9})
+                                'rise_fall': 20e-9})
     gate_chan = Instance((str, LogicalMarkerChannel))
     frequency = Float(0.0).tag(
         desc='modulation frequency of the channel (can be positive or negative)')
+    cnot_impl = Instance(str)
```
Comment (grahamrow) [inline QGL/Channels.py]: Should there be a default implementation?

File: QGL/PulsePrimitives.py [hunk 2]
```diff
@@ -716,14 +716,18 @@ def CNOT_simple(source, target, **kwargs):
...
-    cnot_impl = globals()[config.cnot_implementation]
-except:
-    raise NameError("A valid CNOT implementation was not defined [{}]".format(config.cnot_implementation))
 
 @_memoize
 def CNOT(source, target, **kwargs):
-    return cnot_impl(source, target, **kwargs)
+    channel = ChannelLibraries.EdgeFactory(source, target)
+    if not channel.cnot_impl:
+        print('Warning: CNOT implementation undefined for {}'.format(channel.label))
```
Comment (grahamrow) [inline QGL/PulsePrimitives.py]: Here we assume CR, but don't tell the user that's what we've chosen for them? I'm not sure what the best strategy is.

Comment (dieris) [reply QGL/PulsePrimitives.py]: I agree, we should at least print what the default choice is. My guess is to leave CR as the default, being the most common choice for now

File: QGL/Channels.py (hunk 1 above)
Comment (dieris) [reply QGL/Channels.py]: I'm not sure, I don't have a preference here. But somehow I broke the tests by merging from master...

**Comment by dieris** [comment]
Replaced by https://github.com/BBN-Q/QGL/pull/233

**Pull request closed without merging**